pkg/k8s/init.go

```diff
@@ -134,17 +135,32 @@ func useNodeCIDR(n *nodeTypes.Node) {
 // Init initializes the Kubernetes package. It is required to call Configure()
 // beforehand.
 func Init(conf k8sconfig.Configuration) error {
-	k8sRestClient, closeAllDefaultClientConns, err := createDefaultClient()
+	restConfig, err := CreateConfig()
+	if err != nil {
+		return fmt.Errorf("unable to create k8s client rest configuration: %s", err)
+	}
+	closeAllDefaultClientConns := setDialer(restConfig)
+	// Use the same http client for all k8s connections. It does not matter that
+	// we are using a restConfig for the HTTP client that differs from each
+	// individual client since the rest.HTTPClientFor only does not use fields
+	// that are specific for each client, for example:
+	// restConfig.ContentConfig.ContentType.
+	httpClient, err := rest.HTTPClientFor(restConfig)
```

Comment:
Should we stash this httpClient somewhere global? 

If the need arises to instantiate another K8s resource, like a client, somewhere else in code, having this shared client available will help us not spam more clients across the code base. 

This is a moot point if we only expect decoupled code to just go and grab the Cilium and K8s clients from our k8s package and not create any kind of K8s clients themselves.

Reply:
> This is a moot point if we only expect decoupled code to just go and grab the Cilium and K8s clients from our k8s package and not create any kind of K8s clients themselves.

I think the expectation is that any Cilium code will import `pkg/k8s` to use the "blessed" clients. Anything else should be avoided.